test/utils.dart

```diff
@@ -146,50 +146,123 @@ Future<void> testBenchmarks({required bool useTallStyle}) async {
 void _testFile(TestFile testFile) {
   group(testFile.path, () {
     for (var formatTest in testFile.tests) {
-      // Collect all of the outputs that are specific to a single version.
-      // We'll run those tests at only that version. If there is is an output
-      // with no version specified, then we'll run it on all of the remaining
-      // supported versions.
-      var specificTestedVersions = {
-        for (var output in formatTest.outputs)
-          if (output.version case var version?) version,
-      };
-
-      // Run it for each of the expected outputs.
-      for (var output in formatTest.outputs) {
-        // If the output is version-specific, then only test on that version.
-        if (output.version case var version?) {
+      if (testFile.isTall) {
+        Map<Version, TestEntry> testedVersions;
+        switch (formatTest) {
+          case UnversionedFormatTest(:var output):
+            // This output is unversioned, so test at the lowest and highest
+            // supported versions. If it formats the same on those, we assume
+            // it does for all of the intermediate versions too.
+            testedVersions = {
+              _testedTallVersions.first: output,
+              _testedTallVersions.last: output,
+            };
+          case VersionedFormatTest(:var outputs):
+            testedVersions = _versionedTestEntries(outputs);
+        }
```

Comment:
Can this be a switch expression?

Reply:
Good call. Did it for both switches here.